programs/ssl/ssl_client2.c

```diff
@@ -1158,7 +1211,27 @@ int main( int argc, char *argv[] )
     mbedtls_ctr_drbg_context ctr_drbg;
     mbedtls_ssl_context ssl;
     mbedtls_ssl_config conf;
+
+#if defined(MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL)
+
+#if defined(MBEDTLS_ECP_C)
+    mbedtls_ecp_group_id named_groups_list[MAX_NAMED_GROUPS]; /* list of named groups */
+    mbedtls_ecp_group_id key_share_named_groups_list[MAX_NAMED_GROUPS]; /* list of named groups for key share*/
+    char *start;
+#endif /* MBEDTLS_ECP_C */
+
+#if defined(MBEDTLS_ZERO_RTT)
+    char early_data[] = "early data test";
+#endif /* MBEDTLS_ZERO_RTT */
+
+#if defined(MBEDTLS_SSL_NEW_SESSION_TICKET)
+    mbedtls_ssl_ticket ticket;
+#endif /* MBEDTLS_SSL_NEW_SESSION_TICKET */
+
+#else 
     mbedtls_ssl_session saved_session;
```

Comment:
IIUC this should be enabled even if `MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL` is set, because ultimately we want the library to build and be usable with both TLS 1.2 and TLS 1.3 enabled.